jcl/src/java.base/share/classes/openj9/internal/foreign/abi/InternalDowncallHandler.java

```diff
@@ -123,7 +125,8 @@ void clear() {
 		}
 	}
 
-	private final ThreadLocal<HeapArgInfo> heapArgInfo;
+	private static final ThreadLocal<Deque<HeapArgInfo>> heapArgInfo = ThreadLocal.withInitial(ArrayDeque::new);
+	private static final ThreadLocal<Boolean> flagCreateHeapArgInfo = ThreadLocal.withInitial(()->Boolean.TRUE);
```

Comment:
Please add a comment either here or in `runNativeMethod` describing the behavior of these two fields.